src/components/MultiSelect/MultiSelectDialog.tsx

```diff
@@ -25,30 +24,73 @@ export const MultiSelectDialog = chakra(
   forwardRef<HTMLDivElement, React.PropsWithChildren<MultiSelectDialogProps>>(
     (props, ref) => {
       const {
+        helperText,
         id,
         isBlockElement,
         isDefaultOpen,
+        isSearchable,
+        defaultItemsVisible,
         items,
-        labelText,
+        listOverflow,
+        buttonText,
+        closeOnBlur,
         onApply,
         onChange,
         onClear,
         onMixedStateChange,
         selectedItems,
+        showLabel,
+        showAll,
         width,
         ...rest
       } = props;
 
-      // Use NYPL Breakpoints hook to check screen size
-      const { isLargerThanMobile } = useNYPLBreakpoints();
-
       // Control the open or closed state of the MultiSelect.
-      const [isOpen, setIsOpen] = useState(isDefaultOpen);
+      const [isOpen] = useState(isDefaultOpen);
+      const [itemsList, setItemsList] = useState(items);
+      const [isDropDownOpen, setIsDropDownOpen] = useState(false);
+      const [viewAllLabel, setViewAllLabel] = useState("");
+      const [listHeight, setListHeight] = useState("");
+      const [listItemsCount, setListItemsCount] = useState(defaultItemsVisible);
+      const [defaultCheckboxes, setDefaultCheckboxes] = useState("none");
+      const [defaultChildCheckboxes, setDefaultChildCheckboxes] =
+        useState("none");
+      //const [prevIsOpen, setPrevIsOpen] = React.useState(isDropDownOpen);
+
+      React.useEffect(() => {
+        const calculateListHeight = () => {
+          if (listOverflow === "scroll") {
+            setViewAllLabel("View all");
+            setListHeight("260px");
+            setListItemsCount(defaultItemsVisible);
+          } else {
+            setViewAllLabel("View less");
+            setListHeight("");
+            setListItemsCount(items.length);
+          }
+        };
+
+        calculateListHeight();
```

Comment:
Because you only call this function in this `useEffect`, you can remove it and call the code inside `calculateListHeight` directly. In other words, remove the function and this line but keep the code inside of it:
```
useEffect(() => {
          if (listOverflow === "scroll") {
            setViewAllLabel("View all");
            setListHeight("260px");
            setListItemsCount(defaultItemsVisible);
          } else {
            setViewAllLabel("View less");
            setListHeight("");
            setListItemsCount(items.length);
          }

        if (listItemsCount === defaultItemsVisible) {
          displayDefaultItems();
        } else {
          setItemsList(items);
        }
      }, []);
```